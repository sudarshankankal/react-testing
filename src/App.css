.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.flex {
  display: flex;
  flex-direction: row;
  padding: 15px;
}

.box {
  /* border:1px solid grey; */
  padding:10px;
}

.postItem:hover{
  border:1px solid grey;
  box-shadow: 5px 5px 5px 5px #888888;
  cursor: pointer;
  padding:5px;
  background-color:antiquewhite
}

.postDetail{
  border:1px solid grey;
  box-shadow: 5px 5px 5px 5px #888888;
  cursor: pointer;
  padding:5px;
  background-color:antiquewhite;
  margin:10px
}

.fullWidth{
  width:100%;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
