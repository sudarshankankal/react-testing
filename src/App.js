import "./App.css";
import PostList from "./components/PostList";
import { StoreProvider } from "./components/StoreProvider";
import PostDetail from "./components/PostDetail";
// import FetchData from "./components/FetchData";
import LandingPage from "./pages/LandingPage";
import Nav from "./components/Nav";

function App() {
  return (
    <div>
      <Nav/>
      <LandingPage/>
     
      {/* <StoreProvider>
        <div className="flex">
          <div className="box">
            <PostList />
          </div>
          <div className="box">
            <PostDetail />
          </div>
        </div>
      </StoreProvider> */}
    </div>
  );
}

export default App;
