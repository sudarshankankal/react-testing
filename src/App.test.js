import React from 'react'
import App from './App'
import { mount, shallow } from 'enzyme'

describe('Given App component',() => {

   test("save snapshot", () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot();
  })

  test("Nav is rendered properly", () => {
    const wrapper = shallow(<App/>);
    const nav = wrapper.find("Nav").exists();
    expect(nav).toBe(true);
  })

  test("Landing Page is rendered properly", () => {
    const wrapper = shallow(<App/>);
    const landingPage = wrapper.find("LandingPage").exists();
    expect(landingPage).toBe(true);
  })

})

