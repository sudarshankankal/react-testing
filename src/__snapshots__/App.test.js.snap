// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given App component save snapshot 1`] = `
<div>
  <Nav />
  <LandingPage />
</div>
`;
